Chore: (Docs) Intro to Storybook changes for 6.4 (#545)
chromaui/learnstorybook.com: content/intro-to-storybook/angular/en/screen.md

commit: 'b5f0de2'
---

We've concentrated on building UIs from the bottom up, starting small and adding complexity. Doing so has allowed us to develop each component in isolation, figure out its data needs, and play with it in Storybook. All without needing to stand up a server or build out screens!

In this chapter, we continue to increase the sophistication by combining components in a screen and developing that screen in Storybook.

## Nested container components

> As our app is straightforward, the screen we’ll build is pretty trivial, simply wrapping the `TaskListComponent` (which supplies its own data via ngxs) in some layout and pulling a top-level `error` field out of our store (let's assume we'll set that field if we have some problem connecting to our server).

Reviewer: ```suggestion
As our app is straightforward, the screen we’ll build is pretty trivial, simply wrapping the `TaskList` component (which supplies its own data via ngxs) in some layout and pulling a top-level `error` field out of our store (let's assume we'll set that field if we have some problem connecting to our server).
```

I'm going to stop correcting these. The change to remove "Component" from component names was made in a file above, but applied inconsistently. Fwiw, I agree with the change.

Author: TaskListComponent, TaskComponent, and so on are removed from it. Thanks for this

  />
</video>

## Interactive stories

> So far, we've been able to build a fully functional application from the ground up, starting from a simple component up to a screen, continuously testing each change using our stories. But with each story added, we would also need to check each variation manually to ensure the UI doesn't break, which requires extra work.

Reviewer 2: ```suggestion
So far, we've been able to build a fully functional application from the ground up, starting from a simple component up to a screen and continuously testing each change using our stories. But each new story also requires a manual check on all the other stories to ensure the UI doesn't break. That's a lot of extra work.
```

Author: Fixed with the latest commit for all frameworks.

> Couldn't we automate this workflow and interact with our components automatically?

Author: Fixed with the latest commit for all frameworks.

> Storybook's [`play`](https://storybook.js.org/docs/angular/writing-stories/play-function) function allows us to do just that. They are small snippets of code that run after the story renders. Ideal for scenarios such as ours where we would like to see what happens to our UI when we update our tasks. Since they rely on using framework-agnostic DOM APIs, we can write stories with the play function to interact with the UI and automate human behavior no matter the framework we use.

Author: Fixed with the latest commit for all frameworks.

> Storybook's [`play`](https://storybook.js.org/docs/6.4/angular/writing-stories/play-function) function allows us to do just that. They are small snippets of code that run after the story renders. Ideal for scenarios such as ours where we would like to see what happens to our UI when we update our tasks. Since they rely on using framework-agnostic DOM APIs, we can write stories with the play function to interact with the UI and automate human behavior no matter the framework we use.

Reviewer: If these are changes _for_ `6.4`, I don't think the version should be in the URL, as the correct reference would be "latest" when these are published.

Author: This is provisional so that someone doesn't ask why is it pointing to a 404.

import { moduleMetadata, Meta, Story } from '@storybook/angular';

+ import { fireEvent, within } from '@storybook/testing-library';

import { CommonModule } from '@angular/common';

import { PureInboxScreenComponent } from './pure-inbox-screen.component';
import { TaskModule } from './task.module';

import { Store, NgxsModule } from '@ngxs/store';
import { TasksState } from '../state/task.state';

export default {
  component:PureInboxScreenComponent,
  decorators: [
    moduleMetadata({
      imports: [CommonModule,TaskModule,NgxsModule.forRoot([TasksState])],
      providers: [Store],
    }),
  ],
  title: 'PureInboxScreen',
} as Meta;

const Template: Story = (args) => ({
  props: args,
});

export const Default = Template.bind({});

export const Error = Template.bind({});
Error.args = {
  error: true,
};

+ export const WithInteractions = Template.bind({});
+ WithInteractions.play = async ({ canvasElement }) => {
+   const canvas = within(canvasElement);
+   // Simulates pinning the first task
+   await fireEvent.click(canvas.getByLabelText("pinTask-1"));
+   // Simulates pinning the third task
+   await fireEvent.click(canvas.getByLabelText("pinTask-3"));
+ };
```

> Check your newly created story, and click the `Interactions` panel, and you'll see the following.

Author: Fixed with the latest commit for all frameworks.

<video autoPlay muted playsInline loop>
  <source
    src="/intro-to-storybook/storybook-interactive-stories-play-function.mp4"
    type="video/mp4"
  />
</video>

> As you've seen, we're able to interact with our UI and quickly check how it responds if we update our tasks, ensuring it stays consistent. All without needing to spin up a testing environment or add additional packages.

Reviewer 2: ```suggestion
The play function allows us to interact with our UI and quickly check how it responds if we update our tasks. That keeps the UI consistent at no extra manual effort. All without needing to spin up a testing environment or add additional packages.
```

Author: Fixed with the latest commit for all frameworks.

> As you've seen, we're able to interact with our UI and quickly check how it responds if we update our tasks, ensuring it stays consistent, all of this without needing to spin up a testing environment or add additional packages.

Reviewer: ```suggestion
As you've seen, we're able to interact with our UI and quickly check how it responds if we update our tasks, ensuring it stays consistent. All without needing to spin up a testing environment or add additional packages.
```

[**Component-Driven Development**](https://www.componentdriven.org/) allows you to gradually expand complexity as you move up the component hierarchy. Among the benefits are a more focused development process and increased coverage of all possible UI permutations. In short, CDD helps you build higher-quality and more complex user interfaces.

> We’re not done yet - the job doesn't end when the UI is built, and we also need to ensure that it remains durable over time.

Reviewer 2: ```suggestion
We’re not done yet - the job doesn't end when the UI is built. We also need to ensure that it remains durable over time.
```

Revert back, the grammar isn't quite right.

Author: Fixed with the latest commit for all frameworks.
